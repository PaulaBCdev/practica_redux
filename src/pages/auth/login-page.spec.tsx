import { render } from "@testing-library/react";
import LoginPage from "./login-page";
import { Provider } from "react-redux";

describe("LoginPage", () => {
  const state = {
    auth: false,
    ads: {
      loaded: false,
      data: [],
    },
    tags: {
      loaded: false,
      data: [],
      error: null,
    },
    filters: {
      name: "",
      sale: null,
      price: null,
      tags: [],
      maxPrice: 0,
    },
    ui: {
      pending: false,
      error: null,
    },
  };

  const renderComponent = () =>
    render(
      <Provider
        store={{
          getState: () => state,
          // @ts-expect-error: subscribe
          subscribe: () => {},
          // @ts-expect-error: dispatch
          dispatch: () => {},
        }}
      >
        <LoginPage />
      </Provider>,
    );

  test("should render", () => {
    const { container } = renderComponent();

    expect(container).toMatchSnapshot();
  });
});
