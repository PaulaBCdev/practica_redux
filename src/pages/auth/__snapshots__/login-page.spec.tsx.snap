// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LoginPage > should render 1`] = `
<div>
  <div
    class="layout-page"
  >
    <h2
      class="layout-title"
    >
      Login
    </h2>
    <div
      class="login-page"
    >
      <form
        class="login-form"
      >
        <div
          class="form-field"
        >
          <label
            class="form-field-label login-label"
          >
            Email
          </label>
          <input
            autocomplete="off"
            class="form-field-input login-input"
            name="email"
            type="email"
            value=""
          />
        </div>
        <div
          class="form-field"
        >
          <label
            class="form-field-label login-label"
          >
            Password
          </label>
          <input
            autocomplete="off"
            class="form-field-input login-input"
            name="password"
            type="password"
            value=""
          />
        </div>
        <button
          class="sc-bXdtCk vVNLT login-btn"
          disabled=""
          type="submit"
        >
          Login
        </button>
        <div
          class="remember-me-check"
        >
          <input
            id="rememberMe"
            name="rememberMe"
            type="checkbox"
          />
          <label>
            Remember me
          </label>
        </div>
      </form>
    </div>
  </div>
</div>
`;
